### PR TITLE
Float: Introduce floating point api infinite

### PR DESCRIPTION
**Pull request by Incarnation-p-lee**
* Introduce new float api float_is_inf.
* Add test case for half, single and double floating point.
* Add inf to interface.

Signed-off-by: Pan Li <pan2.li@intel.com>

### CODE REVIEW

File: lib/float/inf.sail [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+
+$ifndef _FLOAT_INF
+$define _FLOAT_INF
+
+$include <float/common.sail>
+
+val      float_is_inf : fp_bits -> bool
+function float_is_inf (op) = {
+  let struct {_, exp, mantissa} = float_decompose(op);
+  let is_nan = exp == sail_ones(length(exp))
```
Comment (jordancarlin) [inline lib/float/inf.sail]: Should probably be `is_inf` instead of `is_nan`

Comment (Incarnation-p-lee) [reply lib/float/inf.sail]: Nice catch, thanks and updated!

File: test/float/inf_test.sail [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+/* SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.             */
+/*==========================================================================*/
+
+default Order dec
+
+$include <prelude.sail>
+$include <float/inf.sail>
+
+function test_float_is_inf() -> unit = {
+  /* Half flating point */
```
Comment (Alasdair) [inline test/float/inf_test.sail]: Typo in the comments here

Comment (Incarnation-p-lee) [reply test/float/inf_test.sail]: Oops, let me update soon and file another PR for other test files.

Comment (Incarnation-p-lee) [reply test/float/inf_test.sail]: Updated.

**Comment by Alasdair** [comment]
Sorry for taking a while with these, I've been rather busy the past week.

**Comment by Incarnation-p-lee** [comment]
> Sorry for taking a while with these, I've been rather busy the past week.

Thanks. Never mind and take your time, ;)!

**Pull request merged**